### PR TITLE
Measure distance

### PR DESCRIPTION
**Pull request by ram-sharan25**
### Measure distance on the map.
Changes 
- Updated readme.
- added `webpack-dev-server` package to support Live Reloading.
- Added a new measure icon to the map toolbar.
- Implemented event listeners for mouse clicks (to set the anchor) and mouse movements (to draw the line and update distance).
- Real-time calculation and rendering of the distance label above the measurement line.

### CODE REVIEW

**Comment by paridhi-parajuli** [comment]
It would be better if you could change the color of the text (showing the measurement), because its hard to see over the raster. May be try putting the text over a box with white background.
@ram-sharan25 @sanzog03

**Comment by ram-sharan25** [comment]
Fixes :

- Fixed bugs related to the eraser tool.
- Updated the measurement label's position to the tip of the line, ensuring visibility regardless of the line's length or position. This also resolved the issue where labels weren't shown for very short distances.Previously for shorter distance line length was not long enough to anchor the label on the map. 
- Changed the measurement label color from blue to white with a black border, aligning it with other labels on the map for consistency.
@paridhi-parajuli  @amarouane-ABDELHAK  @sanzog03

File: emit-ch4plume-v1/src/index.js [hunk 1]
```diff
@@ -25,525 +25,772 @@ const markerProps = new Object();
...
+    return this.container;
+  }
+  onRemove() {
+    this.container.parentNode.removeChild(this.container);
+    this.map = undefined;
+  }
+}
+class ClearDistancePoints {
+  onClick() {
+    distancePoints.features.splice(0, distancePoints.features.length);
```
Comment (sanzog03) [inline emit-ch4plume-v1/src/index.js:195]: Is this used to clear the features list? if yes we could rather do .length = 0 for a constant time operation.

File: emit-ch4plume-v1/src/index.js [hunk 2]
```diff
@@ -25,525 +25,772 @@ const markerProps = new Object();
...
+      '<div id="clear-icon-main">' +
+      "<button>" +
+      '<span id="clear-icon" class="mapboxgl-ctrl-icon" aria-hidden="true" title="Measure Tool">' +
+      '<i class="fa-solid fa-eraser"></i>' +
+      "</span>" +
+      "</button>" +
+      "</div>";
+    return this.container;
+  }
+  onRemove() {
```
Comment (sanzog03) [inline emit-ch4plume-v1/src/index.js:219]: This seems to be reused in many other control classes. could create an abstract class with this `onRemove()` reused definition. Inheriting that would remove the need for this one in  `ClearDistancePoints` control class. More DRY the better.

File: emit-ch4plume-v1/src/index.js [hunk 3]
```diff
@@ -25,525 +25,772 @@ const markerProps = new Object();
...
+    this.container.parentNode.removeChild(this.container);
+    this.map = undefined;
+  }
+}
+class ClearDistancePoints {
+  onClick() {
+    distancePoints.features.splice(0, distancePoints.features.length);
+     measureLine.features.splice(0, measureLine.features.length);
```
Comment (sanzog03) [inline emit-ch4plume-v1/src/index.js:196]: same as above

File: emit-ch4plume-v1/src/index.js [hunk 4]
```diff
@@ -25,525 +25,772 @@ const markerProps = new Object();
...
+     distanceLabel.features.splice(0, distanceLabel.features.length);
```
Comment (sanzog03) [inline emit-ch4plume-v1/src/index.js:197]: same as above

File: emit-ch4plume-v1/README.md [hunk 5]
```diff
@@ -1,5 +1,65 @@
...
+
+- [Node](http://nodejs.org/) v20 (To manage multiple node versions we recommend [nvm](https://github.com/creationix/nvm))
+- [npm](https://www.npmjs.com/) Package manager
+- [Yarn](https://yarnpkg.com/) Package manager
+
+## Install Application Dependencies
+
+If you use [`nvm`](https://github.com/creationix/nvm), activate the desired Node version:
+
+`nvm install`
```
Comment (sanzog03) [inline emit-ch4plume-v1/README.md:15]: add a .nvmrc file with appropriate version. 
And add `nvm use` in the readme

**Comment by sanzog03** [comment]
@ram-sharan25 The ruler icon when selected and unselected, looks very similar.  Need to update that to make them clear. 
@abarciauskas-bgse Also a thought, when the scale tool is active we could disable or even hide the datetime filter and search. It might help in further eliminating the edge cases to be handled.

**Comment by abarciauskas-bgse** [comment]
@sanzog03 did you mean to tag me in this PR?

**Comment by sanzog03** [comment]
sorry it was a mistake!

**Comment by ram-sharan25** [comment]
# New Changes

- Added `.nvmrc` to specify Node.js version.
- Updated README to include usage of `.nvmrc`.
- Refactored class definition in `MapControls` for better organization.
- Added feature to toggle map scale between kilometers and miles for both measurement mode and the map overall.
- Disabled measure mode on double-click to improve user interaction.
- Disabled search and date range pickers when measure mode is active.
- Updated measure icon to make it more prominent, with a clear visual indication when switching between modes.
@sanzog03  @amarouane-ABDELHAK

**Comment by amarouane-ABDELHAK** [comment]
Looks good to me. Thank you

File: emit-ch4plume-v1/README.md [hunk 6]
```diff
@@ -1,5 +1,69 @@
...
+`cp .env.local-sample .env`
+
+Replace `xxxxxxx` with keys and token.
+
+### Get Data
+
+To set the environment variables from the `.env` file, add export in front of each variable. 
+For example:
+
+`export MAP_STYLE=“xxxxxxxxxxxx”`
```
Comment (amarouane-ABDELHAK) [inline emit-ch4plume-v1/README.md:45]: Can you replace this by running 
`node --env-file .env update_data.js`
When running this command the developer should not have to export the env vars

**Review by amarouane-ABDELHAK — APPROVED** [review]

**Pull request merged**